### PR TITLE
executor: make the format of the DB field in slow ... (#18389)

### LINKED ISSUE #17846: make slow logs be compatible with `pt-query-digest`
## Development Task
Some users like to use `pt-query-digest` to analyze slow queries, but our slow logs are not compatible with `pt-query-digest` perfectly.

We list all issues here:

- [x] cannot get `user/host_max` fields #18162
- [ ] cannot get `db_max` fields #18389

Here is [a Chinese document](https://docs.google.com/document/d/1nb-7DMr8ne3jsop2Jl0Nn9dxU9Usf4UecjQBK_TLKpA/edit?usp=sharing) about these issues.

#### 1) `pt-query-digest` cannot get `user/host_max` fields
The reason is that `TiDB` and `MySQL` have different format in these fields.

TiDB's format: `# User: {user_name}@{host_address}`
For example: `# User: root@127.0.0.1`

MySQL: `# User@Host: {user_name}[{user_name}] @ {host_name} [{host_address}]`
For example: `# User@Host: root[root] @ localhost [127.0.0.1]`

#### 2) `pt-query-digest` cannot get `db_max` fields
In TiDB's slow logs, there is only one SQL after comments:
image
And the database for this SQL is identified by the comment `#DB`.

But in MySQL, there may be multiple SQLs, and the database used for the corresponding query is identified by `use DB`.
image

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #18389 to release-4.0

---

### What problem does this PR solve?

Issue Number: close #17846 

Problem Summary: make the format of the DB field in slow logs be compatible with MySQL

### What is changed and how it works?

Print `use {DB}` statements into the slow log.

Please see #17846 for more details.

### Check List 

Tests 

- Unit test

### Release note 

- executor: make the format of the DB field in slow logs be compatible with MySQL

### CODE REVIEW

File: sessionctx/variable/session_test.go [hunk 1]
```diff
@@ -174,8 +174,13 @@ func (*testSessionSuite) TestSlowLogFormat(c *C) {
 
 	var memMax int64 = 2333
 	var diskMax int64 = 6666
+<<<<<<< HEAD
```
Comment (lzmhhh123) [inline sessionctx/variable/session_test.go:177]: @qw4990 Pls resolve the conflists.

**Review by zz-jason — COMMENTED** [review]
@qw4990 please resolve conflicts and fix CI.

**Comment by qw4990** [comment]
Wait for https://github.com/pingcap/tidb/pull/18379.

**Comment by qw4990** [comment]
Wait for https://github.com/pingcap/tidb/pull/18164.

**Review by zz-jason — APPROVED** [review]
LGTM

**Pull request merged**